src/deepsparse/transformers/eval_downstream.py

```diff
@@ -75,36 +76,97 @@
 from datasets import load_dataset, load_metric  # isort: skip
 
 
-def perplexity_eval(args, batch_size=16, dataset_name="openai_humaneval"):
-    if args.max_samples:
-        batch_size = min(batch_size, args.max_samples)
+def perplexity_eval(args, dataset_name="openai_humaneval"):
 
-    dataset = load_dataset(dataset_name)["test"]
+    if dataset_name in ["wikitext", "c4"]:
+        if args.kwargs is None:
+            kwargs = {}
+        else:
+            kwargs = json.loads(args.kwargs)
+        dataset = _process_concatenated_datasets(
+            dataset_name,
+            args.model_path,
+            args.max_sequence_length,
+            kwargs,
+        )
+        # Set perplexity computation to accumulate negative log-likelihood across
+        # sections
+        accumulate = True
+    else:
+        dataset = load_dataset(dataset_name, split="test")
+        accumulate = False
 
+    # We'll use the text generation pipeline to generate a single token.
+    # Along with the token, it returns the logits for input sequence
     text_generation = Pipeline.create(
         task="text-generation",
         model_path=args.model_path,
         engine_type=args.engine,
         num_cores=args.num_cores,
         sequence_length=args.max_sequence_length,
         max_generated_tokens=1,
+        trust_remote_code=args.trust_remote_code,
+        batch_size=args.batch_size,
```

Comment:
```suggestion
```

@anmarques this causes the error.

Currently, we cannot run LLM models in the engine with `batch_size != 1`. This is the batch size of the engine, to be clear, not the batch size of the evaluation script.

Now:
The engine inputs at batch 16, get split into a input of a batch size 16 and this gives you the error.


**After you accept the changes:**
The engine inputs at batch 16, will be properly split into 16 separate inputs of batch size 1 band each batch will be processed by the engine that assumes batch_size=1

Reply:
Addressed